charts/skypilot/templates/api-deployment.yaml

```diff
@@ -0,0 +1,191 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: skypilot-api-server
+  namespace: {{ .Release.namespace }}
+spec:
+  replicas: {{ .Values.apiService.replicas }}
```

Comment:
We may want to have a comment here or in the values that `replicas > 1` is not tested yet.